base/primes.jl

```diff
@@ -32,16 +32,17 @@ primesmask(n::Integer) = n <= typemax(Int) ? primesmask(Int(n)) :
 
 primes(n::Union(Integer,AbstractVector{Bool})) = find(primesmask(n))
 
-# Miller-Rabin for primality testing:
+const PRIMES = primes(2^16)
+
+# Small precomputed primes + Miller-Rabin for primality testing:
 #     http://en.wikipedia.org/wiki/Miller–Rabin_primality_test
 #
 function isprime(n::Integer)
-    n == 2 && return true
-    (n < 2) | iseven(n) && return false
+    (n < 3 || iseven(n)) && return n == 2
+    n <= 2^16 && return length(searchsorted(PRIMES,n)) == 1
```

Comment:
maybe `searchsortedlast(PRIMES,n,1,length(PRIMES),Forward) > 0`?  `searchsorted` seems suboptimal for this purpose since it performs two binary searches rather than one (to find the first and last occurrences of `n`.

Probably we should add a function to `sort.jl` for this:
```jl
sortedcontains(v::AbstractVector, x, o::Ordering=Forward) = searchsortedlast(v, x, 1, length(v), o) > 0
```